core/chains/evm/monitor/balance_test.go

```diff
@@ -254,3 +256,24 @@ func Test_ApproximateFloat64(t *testing.T) {
 		})
 	}
 }
+
+func callbackOrTimeout(t testing.TB, msg string, callback func(), durationParams ...time.Duration) {
+	t.Helper()
+
+	duration := 100 * time.Millisecond
+	if len(durationParams) > 0 {
+		duration = durationParams[0]
+	}
+
+	done := make(chan struct{})
+	go func() {
+		callback()
+		close(done)
```

Comment:
Pre-existing, but to avoid deadlock in the case of a panicking `callback()`:
```suggestion
		defer close(done)
		callback()
```